btr-api/src/btr_api/services/auth.py

```diff
@@ -59,66 +65,136 @@ def __init__(self, app: Flask = None):
+            raise ExternalServiceException(
+                HTTPStatus.GATEWAY_TIMEOUT,
+                [{"message": "Unable to get service account token.", "reason": err.with_traceback(None)}],
+            ) from err
         except Exception as err:  # noqa: B902
-            self.app.logger.debug('SSO SVC connection failure: %s', err.with_traceback(None))
-            raise ExternalServiceException(HTTPStatus.SERVICE_UNAVAILABLE,
-                                           [{'message': 'Unable to get service account token.',
-                                             'reason': err.with_traceback(None)}]) from err
+            self.app.logger.debug("SSO SVC connection failure: %s", err.with_traceback(None))
+            raise ExternalServiceException(
+                HTTPStatus.SERVICE_UNAVAILABLE,
+                [{"message": "Unable to get service account token.", "reason": err.with_traceback(None)}],
+            ) from err
 
     def get_authorization_header(self, request) -> str:
         """Gets authorization header from request."""
-        authorization_header = request.headers.get('Authorization', None)
+        authorization_header = request.headers.get("Authorization", None)
         if not authorization_header:
-            error = f'Missing authorization header: {request.headers}'
-            self.app.logger.debug('Cannot find authorization_header in request.')
+            error = f"Missing authorization header: {request.headers}"
+            self.app.logger.debug("Cannot find authorization_header in request.")
             raise AuthException(error=error, status_code=HTTPStatus.UNAUTHORIZED)
 
         return authorization_header
 
+    # TODO: Add caching
+    def product_authorizations(self, request, account_id: str) -> bool:
+        """Authorize the user for access to the service."""
+        if not account_id:
+            error = "Missing account_id"
+            self.app.logger.debug(error)
+            request.product_resp = {}
+            return True
+        try:
+            auth_token = self.get_authorization_header(request)
+            # make api call
+            headers = {"Authorization": auth_token}
+            auth_url = f"{self.svc_url}/orgs/{account_id}/products?include_hidden=true"
+            self.app.logger.debug(auth_url)
+            resp = requests.get(url=auth_url, headers=headers, timeout=self.timeout)
+
+            if resp.status_code >= HTTPStatus.INTERNAL_SERVER_ERROR:
+                error = f"{resp.status_code} - {str(resp.json())}"
+                self.app.logger.debug("Invalid response from auth-api: %s", error)
+                request.product_resp = {}
+
+            if resp.status_code != HTTPStatus.OK:
+                error = f"Unauthorized access to products {account_id}"
+                self.app.logger.debug(error)
+                request.product_resp = {}
+
+            request.product_resp = resp
```

Comment:
I see your adding stuff to the request object and then passing that around, but it's hard to follow.

I don't think it adds a lot of value to add stuff to the request obj and pass it around everywhere (especially when its renamed to other things), but we can discuss

Reply:
also, it looks like it will override it even on the error, if any of the error response ifs is triggered, it will set this value to empty, then after the ifs, it will set it to resp itself, is this intended